src/Configs/Params.php

```diff
@@ -12,24 +12,27 @@ protected function calcValues(array $sources): array
         return $this->pushEnvVars(parent::calcValues($sources));
     }
 
-    protected function pushEnvVars($vars): array
+    protected function pushEnvVars(array $vars): array
     {
-        $env = $this->builder->getConfig('dotenv')->getValues();
-        if (!empty($vars)) {
-            foreach ($vars as $key => &$value) {
-                if (is_array($value)) {
-                    foreach (array_keys($value) as $subkey) {
-                        $envKey = $this->getEnvKey($key . '_' . $subkey);
-                        if (isset($env[$envKey])) {
-                            $value[$subkey] = $env[$envKey];
-                        }
-                    }
-                } else {
-                    $envKey = $this->getEnvKey($key);
+        if ($vars === []) {
```

Comment:
```suggestion
        if (empty($vars)) {
```
Very strange way to check if an array is empty :-/ 
https://stackoverflow.com/a/59841707/6757340
Performance is not that important here but still ...

Reply:
Nice benchmark, thanks!